src/dialogs/quick-open/ha-quick-open-dialog.ts

```diff
@@ -0,0 +1,216 @@
+  @property({ attribute: false }) public hass!: HomeAssistant;
+
+  @internalProperty() private _items: QuickOpenItem[] = [];
+
+  @internalProperty() private _entityFilter = "";
+
+  @internalProperty() private _opened = false;
+
+  @internalProperty() private _commandMode = false;
+
+  public async showDialog(params: QuickOpenDialogParams) {
+    this._commandMode = params.commandMode || false;
+    this._opened = true;
+  }
+
+  public closeDialog() {
+    this._opened = false;
+    this._entityFilter = "";
+    fireEvent(this, "dialog-closed", { dialog: this.localName });
+  }
+
+  private _entityFilterChanged(ev: PolymerChangedEvent<string>) {
+    this._entityFilter = ev.detail.value;
+    if (this._entityFilter.startsWith(">")) {
+      this._commandMode = true;
+      this._entityFilter = this._entityFilter.substring(1);
+    } else {
+      this._commandMode = false;
+    }
+
+    let items: QuickOpenItem[];
+    if (this._commandMode) {
+      items = this._generateCommandItems(this._entityFilter);
+    } else {
+      items = this._generateEntityItems(this.hass, this._entityFilter);
+    }
+    this._items = this._filterItems(items, this._entityFilter);
+  }
+
+  private _generateCommandItems(_filter: string): QuickOpenItem[] {
+    const reloadableDomains = componentsWithService(this.hass, "reload").sort();
+
+    return reloadableDomains.map((domain) => ({
+      text:
+        this.hass.localize(`ui.dialogs.quick_open.commands.${domain}`) ||
+        this.hass.localize(
+          "ui.dialogs.quick_open.commands.reload",
+          "domain",
+          domainToName(this.hass.localize, domain)
+        ),
+      domain,
+      onClick: this._executeCommand,
+    }));
+  }
+
+  private _generateEntityItems(
+    hass: HomeAssistant,
+    _filter: string
+  ): QuickOpenItem[] {
+    return Object.keys(hass.states)
+      .map((key) => hass.states[key])
+      .map<QuickOpenItem>(({ entity_id }) => ({
+        text: entity_id,
+        domain: computeDomain(entity_id),
+        onClick: this._entityMoreInfo,
+      }));
+  }
+
+  private _filterItems(
+    items: QuickOpenItem[],
+    filter: string
+  ): QuickOpenItem[] {
+    return items
+      .filter(({ text }) => fuzzySequentialMatch(filter.toLowerCase(), text))
+      .sort((itemA, itemB) => {
+        if (itemA.text < itemB.text) {
+          return -1;
+        }
+        if (itemA.text > itemB.text) {
+          return 1;
+        }
+        return 0;
+      });
+  }
+
+  private async _executeCommand(ev: Event) {
+    const eventData = {
+      domain: (ev.target as any).domain,
+      service: "reload",
+    };
+
+    await this.hass.callService(eventData.domain, "reload");
+
+    fireEvent(this, "hass-service-called", eventData);
+
+    this.closeDialog();
+  }
+
+  private _entityMoreInfo(ev: Event) {
+    ev.preventDefault();
+    fireEvent(this, "hass-more-info", {
+      entityId: (ev.target as any).text,
+    });
+    this.closeDialog();
+  }
+
+  protected render() {
```

Comment:
Please keep the order: public, protected, private. So move this up

Reply:
Will do. OOC, does the frontend repo have linting rules for this? Just want to make sure my vscode extensions aren't overriding HA-specific linters